app/src/main/java/com/example/rickandmorty/domain/search/GetSearchResultUseCase.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.rickandmorty.domain.search
+
+import com.example.rickandmorty.domain.CharacterClient
+
+/**
+ * Class that gets the Data from CharacterClients
+ * and perform functions such as
+ * sorting and filtering
+ **/
+class GetSearchResultUseCase(
+    private val characterClient: CharacterClient,
+) {
+
+    suspend fun execute(queryString: String, page: Int = 1): SearchResult? {
```

Comment:
Same suggestions as above, the number should be in a constant. Since you will probably use in different places you can have it in one common place and reuse it